test/e2e-go/restAPI/restClient_test.go

```diff
@@ -170,14 +170,14 @@ func waitForTransaction(t *testing.T, testClient libgoal.Client, fromAddress, tx
 	}
 	timeoutTime := time.Now().Add(timeout)
 	for {
-		tx, err = testClient.TransactionInformation(fromAddress, txID)
-		if err != nil && strings.HasPrefix(err.Error(), "HTTP 404") {
-			tx, err = testClient.PendingTransactionInformation(txID)
-		}
+		// tx, err = testClient.TransactionInformation(fromAddress, txID)
+		// if err != nil && strings.HasPrefix(err.Error(), "HTTP 404") {
+		tx, err = testClient.ParsedPendingTransaction(txID)
+		// }
 		if err == nil {
 			a.NotEmpty(tx)
 			a.Empty(tx.PoolError)
-			if tx.ConfirmedRound > 0 {
+			if tx.ConfirmedRound != nil && *tx.ConfirmedRound > 0 {
```

Comment:
@algochoi Optional and may require further thought - Have you considered enriching `ParsedPendingTransaction` to require the caller to specify the transaction state?

* I imagine there's 3 policies for returning an error:  
  *  Find any transaction - Return error only when no transaction found (neither pending nor confirmed).
  * Pending only - Return error either when no transaction found or transaction is confirmed.
  * Confirmed only - Return error either when no transaction found or transaction is pending.
* I can imagine the API taking several forms.  The intention is to avoid usage mistakes that are liable to happen when `ConfirmedRound` is not checked.

Reply:
I haven't (there's only 1 use in the current codebase), but I think that could be a good idea for future work.

Reply:
@algochoi Resolving - On 2nd look, I think I'll circle back to the concern should it cause problems later.  

* I can imagine a variety of design debates happening because there's a couple of usage patterns.  Let's leave as is since it's working today.  
* For completeness, when considering `RestClient.PendingTransactionInformationV2` and `RestClient.RawPendingTransactionInformationV2` there's additional use cases.